packages/mapper/src/mapper.ts

```diff
@@ -1682,10 +1727,63 @@ export default class Mapper {
   }
 
   collection_getFullName(collection: Collection): string {
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'getFullName',
+        class: 'Collection',
+        db: collection._database._name,
+        coll: collection._name
+      }
+    );
+
     return `${collection._database._name}.${collection._name}`;
   }
 
   collection_getName(collection: Collection): string {
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'getName',
+        class: 'Collection',
+        db: collection._database._name,
+        coll: collection._name
+      }
+    );
+
     return `${collection._name}`;
   }
+
+  async collection_runCommand(
+    collection: Collection,
+    commandName: string,
+    options?: Record<string, any>
+  ): Promise<any> {
+    if (typeof commandName !== 'string') {
+      throw new Error('commandName must be a string');
```

Comment:
How do you feel about using `assert` here instead of using an Error. Lots of these are look like asserts to me.

Reply:
Sounds good to me!

Reply:
Going to do that in a separate PR cause i have to fix it in other paces too.